### PR TITLE
Updating Docs

### PR DESCRIPTION
**Pull request by BradReesWork**
starting with all the packages in a table

### CODE REVIEW

**Review by acostadon — APPROVED** [review]
I like the content. This is just a preference, but I would have put the detail about each section into an extra column in the table rather than below it.

As for an image above, I was thinking maybe a visual to divide cuGraph into its functionality.
Like a pie with sections for  

- data scientists/python
- C/C++ application integrators
- GNN solution spaces
Not sure that is the right approach just thinking

File: docs/cugraph/source/index.rst [hunk 1]
```diff
@@ -1,15 +1,40 @@
...
+
+
+.. list-table:: RAPIDS Graph covers a range of graph libraries and packages, that includes:
+   :widths: 25 25 25
+   :header-rows: 1
+
+   * - Core
+     - GNNß
+     - Extension
+   * - :abbr:`cugraph (Heavy-weight Python wrapper with lot of guard rails)`
```
Comment (rlratzel) [inline docs/cugraph/source/index.rst:16]: minor typo, but the bigger feedback is that IMO "heavy-weight" has bad connotations, do we want to use that?
```suggestion
   * - :abbr:`cugraph (Python wrapper with lots of guard rails)`
```

File: docs/cugraph/source/index.rst [hunk 2]
```diff
@@ -1,15 +1,40 @@
...
+     - :abbr:`cugraph-ops (GPU-accelerated GNN aggregators and operators)`
+     - :abbr:`cugraph-service (Graph-as-a-service provides both Client and Server packages)`
+   * - :abbr:`pylibcugraph (light-weight Python wrapper no guard rails)`
```
Comment (rlratzel) [inline docs/cugraph/source/index.rst:19]:
```suggestion
   * - :abbr:`pylibcugraph (light-weight Python wrapper with no guard rails)`
```

**Review by rlratzel — CHANGES_REQUESTED** [review]
a couple very minor things, and a question

File: docs/cugraph/source/index.rst (hunk 1 above)
Comment (BradReesWork) [reply docs/cugraph/source/index.rst]: fixed

File: docs/cugraph/source/index.rst (hunk 2 above)
Comment (BradReesWork) [reply docs/cugraph/source/index.rst]: fixed

**Review by rlratzel — APPROVED** [review]

**Pull request merged**